hydromt_wflow/workflows/waterbodies.py

```diff
@@ -482,3 +482,158 @@ def reservoirattrs(gdf, timeseries_fn=None, perc_norm=50, perc_min=20, logger=lo
+            "Dis_avg": "LakeAvgOut",
+        }
+    )
+    # Add maximum volume / no filling of NaNs as assumes then natural lake and not controlled
+    if add_maxstorage:
+        if "Vol_max" in gdf.columns:
+            gdf = gdf.rename(columns={"Vol_max": "LakeMaxStorage"})
+        else:
+            logger.warning(
+                "No maximum storage 'Vol_max' column found, skip adding LakeMaxStorage map."
+            )
+    # Minimum value for LakeAvgOut
+    LakeAvgOut = gdf["LakeAvgOut"].copy()
+    gdf["LakeAvgOut"] = np.maximum(gdf["LakeAvgOut"], 0.01)
+    if "Lake_b" not in gdf.columns:
+        gdf["Lake_b"] = gdf["LakeAvgOut"].values / (gdf["LakeAvgLevel"].values) ** (2)
+    if "Lake_e" not in gdf.columns:
+        gdf["Lake_e"] = 2
+    if "LakeThreshold" not in gdf.columns:
+        gdf["LakeThreshold"] = 0.0
+    if "LinkedLakeLocs" not in gdf.columns:
+        gdf["LinkedLakeLocs"] = 0
+    if "LakeStorFunc" not in gdf.columns:
+        gdf["LakeStorFunc"] = 1
+    if "LakeOutflowFunc" not in gdf.columns:
+        gdf["LakeOutflowFunc"] = 3
+
+    # Check if some LakeAvgOut values have been replaced
+    if not np.all(LakeAvgOut == gdf["LakeAvgOut"]):
+        logger.warning(
+            "Some values of LakeAvgOut have been replaced by a minimum value of 0.01m3/s"
+        )
+
+    # Check if rating curve is provided
+    rating_curves = dict()
+    if len(rating_dict) != 0:
+        # Assume one rating curve per lake index
+        for id in gdf["waterbody_id"].values:
+            if id in rating_dict.keys():
+                df_rate = rating_dict[id]
+                # Prepare the right tables for wflow
+                # Update LakeStor and LakeOutflowFunc
+                # Storage
+                if "volume" in df_rate.columns:
+                    gdf.loc[gdf["waterbody_id"] == id, "LakeStorFunc"] = 2
+                    df_stor = df_rate[["elevtn", "volume"]].dropna(
+                        subset=["elevtn", "volume"]
+                    )
+                    df_stor.rename(columns={"elevtn": "H", "volume": "S"}, inplace=True)
+                    # add to rating_curves
+                    rating_curves[f"lake_sh_{id}"] = df_stor
+                else:
+                    logger.warning(
+                        f"Storage data not available for lake {id}. Using default S=AH"
+                    )
+                # Rating
+                if "discharge" in df_rate.columns:
+                    gdf.loc[gdf["waterbody_id"] == id, "LakeOutflowFunc"] = 1
+                    df_rate = df_rate[["elevtn", "discharge"]].dropna(
+                        subset=["elevtn", "discharge"]
+                    )
+                    df_rate.rename(
+                        columns={"elevtn": "H", "discharge": "Q"},
+                        inplace=True,
+                    )
+                    # Repeat Q for the 365 JDOY
+                    df_q = pd.concat([df_rate.Q] * (366), axis=1, ignore_index=True)
```

Comment:
Do we need a table with a column for each day of the year if these are all the same?
And what happens with in leap years?

Reply:
Yes this is the Wflow format... Also I think they then take care of the leap years (these are like cyclic tables for JDOY). I also tested with wflow.